tests/filtered_relation/tests.py

```diff
@@ -424,3 +585,182 @@ def test_aggregate(self):
+            title='The book by Jane B',
+            editor=cls.editor_b,
+            author=cls.author2,
+        )
+
+        cls.seller1 = BookSeller.objects.create(name="Slamazon")
+        cls.seller2 = BookSeller.objects.create(name="Shwalmart")
+
+        cls.sales_date1 = datetime(2020, 7, 6, 0, 0)
+        cls.sales_date2 = datetime(2020, 7, 7, 0, 0)
+        cls.currency1 = CurrencyCode.objects.create(currency_name="USD")
+        cls.currency2 = CurrencyCode.objects.create(currency_name="EUR")
+
+        CurrencyConversion.objects.create(
+            date=cls.sales_date1,
+            from_currency=cls.currency1,
+            to_currency=cls.currency2,
+            rate=0.40
+        )
+        CurrencyConversion.objects.create(
+            date=cls.sales_date1,
+            from_currency=cls.currency2,
+            to_currency=cls.currency1,
+            rate=2.50
+        )
+        CurrencyConversion.objects.create(
+            date=cls.sales_date2,
+            from_currency=cls.currency1,
+            to_currency=cls.currency2,
+            rate=0.50
+        )
+        CurrencyConversion.objects.create(
+            date=cls.sales_date2,
+            from_currency=cls.currency2,
+            to_currency=cls.currency1,
+            rate=2.00
+        )
+        CurrencyConversion.objects.create(
+            date=cls.sales_date1,
+            from_currency=cls.currency1,
+            to_currency=cls.currency1,
+            rate=1.00
+        )
+        CurrencyConversion.objects.create(
+            date=cls.sales_date2,
+            from_currency=cls.currency1,
+            to_currency=cls.currency1,
+            rate=1.00
+        )
+        CurrencyConversion.objects.create(
+            date=cls.sales_date1,
+            from_currency=cls.currency2,
+            to_currency=cls.currency2,
+            rate=1.00
+        )
+        CurrencyConversion.objects.create(
+            date=cls.sales_date2,
+            from_currency=cls.currency2,
+            to_currency=cls.currency2,
+            rate=1.00
+        )
+
+        BookDailySales.objects.create(
+            book=cls.book1,
+            date=cls.sales_date1,
+            currency_code=cls.currency1,
+            sales=100.00,
+            seller=cls.seller1,
+        )
+        BookDailySales.objects.create(
+            book=cls.book2,
+            date=cls.sales_date1,
+            currency_code=cls.currency2,
+            sales=200.00,
+            seller=cls.seller1,
+        )
+        BookDailySales.objects.create(
+            book=cls.book1,
+            date=cls.sales_date2,
+            currency_code=cls.currency1,
+            sales=50.00,
+            seller=cls.seller2,
+        )
+        BookDailySales.objects.create(
+            book=cls.book2,
+            date=cls.sales_date2,
+            currency_code=cls.currency2,
+            sales=100.00,
+            seller=cls.seller2,
+        )
+
+    def test_with_summation_aggregation(self):
```

Comment:
This test reflects how I am actually using this feature to get more performance out of some queries by improving the join-on logic.